sdk/python/kfp/dsl/databricks/_cluster_op.py

```diff
@@ -0,0 +1,133 @@
+            spec: Specification of the Databricks cluster to create.
+
+        Raises:
+
+            ValueError: if not inside a pipeline
+                        if the name is an invalid string
+                        if no cluster name is provided
+        """
+
+        if not cluster_name:
+            raise ValueError("You need to provide a cluster_name.")
+
+        if not spec:
+            raise ValueError("You need to provide a spec.")
+
+        super().__init__(
+            k8s_resource = {
+                "apiVersion": "databricks.microsoft.com/v1alpha1",
+                "kind": "Dcluster",
+                "metadata": {
+                    "name": cluster_name,
+                },
+                "spec": spec,
+            }, 
+            action = "create",
+            success_condition = "status.cluster_info.cluster_id != ", 
+            attribute_outputs = {
+                "name": "{.status.cluster_info.cluster_id}",
+                "cluster_id": "{.status.cluster_info.cluster_id}",
+                "cluster_name": "{.metadata.name}"
+            },
+            name = name)
+
+    @property
+    def resource(self):
+        """`Resource` object that represents the `resource` property in
+        `io.argoproj.workflow.v1alpha1.Template`.
+        """
+        return self._resource
+
+class DeleteClusterOp(ResourceOp):
+    """Represents an op which will be translated into an Azure Databricks Cluster deletion resource template.
+
+    Example:
+
+        import kfp.dsl.databricks as databricks
+        
+        databricks.DeleteClusterOp(
+            name="deletecluster",
+            cluster_name="test-cluster"
+        )
+    """
+
+    def __init__(self,
+        name: str = None,
+        cluster_name: str = None):
+        """Create a new instance of DeleteClusterOp.
+
+        Args:
+
+            name: the name of the op. It does not have to be unique within a pipeline
+                because the pipeline will generates a unique new name in case of conflicts.
+            cluster_name: the name of the Databricks cluster.
+        
+        Raises:
+
+            ValueError: if not inside a pipeline
+                        if the name is an invalid string
+                        if no cluster name is provided
+        """
+
+        if not cluster_name:
+            raise ValueError("You need to provide a cluster_name.")
+
+        super().__init__(
+            k8s_resource = {
+                "apiVersion": "databricks.microsoft.com/v1alpha1",
+                "kind": "Dcluster",
+                "metadata": {
+                    "name": cluster_name,
+                },
+            }, 
+            action = "delete",
+            name = name)
+
+    @property
+    def resource(self):
+        """`Resource` object that represents the `resource` property in
+        `io.argoproj.workflow.v1alpha1.Template`.
+        """
+        return self._resource
```

Comment:
have you run this code through a linter/formatter?
just curious, as many linters/formatters prefer there to be a line feed at the end of the last line of code

Reply:
I just realized linter wasn't working on my VS Code instance. I managed to make it work. I'll review the feedback from the tool